pkg/ibm_plugin/sdk/tagging.go

```diff
@@ -34,26 +34,27 @@ func (c *CloudClient) attachTag(CRN *string, tags []string) error {
 		ResourceID:   CRN,
 		ResourceType: &userTypeTag,
 	}
+
 	attachTagOptions := c.taggingService.NewAttachTagOptions(
 		[]globaltaggingv1.Resource{*resourceModel},
 	)
 	attachTagOptions.SetTagNames(tags)
 
 	// attach tags with retires.
 	// retry mechanism improves stability and is needed due to possible temporary unavailability of resources, e.g. at time of creation.
-	maxAttempts := 10    // retries number to tag a resource
-	latestResponse := "" // record latest response from inner scope
-	for attempt := 1; attempt <= maxAttempts; attempt += 1 {
-		result, latestResponse, _ := c.taggingService.AttachTag(attachTagOptions)
+	maxAttempts := 10 // retries number to tag a resource
+	var err error
+	for attempt := 1; attempt <= maxAttempts; attempt++ {
+		result, _, err := c.taggingService.AttachTag(attachTagOptions)
 		// tracking all responses from error prone tagging service
-		utils.Log.Printf("Tagging attempt %v: Response: %+v", attempt, latestResponse)
+		utils.Log.Printf("Tagging attempt %v: Error: %v", attempt, err)
 		if !*result.Results[0].IsError {
 			return nil
 		}
 		// sleep to avoid busy waiting
 		time.Sleep(5 * time.Second)
 	}
-	return fmt.Errorf("Failed to tag resource with response:\n %+v", latestResponse)
+	return fmt.Errorf("failed to tag resource : %v", err)
```

Comment:
I believe you're trying to return an error at the end if all the attempts fail, but `err` gets redeclared within the scope of the for loop because of the `:=`. So line 57 actually would just have a blank `err`.

Correct me if I'm wrong though.

Reply:
Ah, good catch..Will update

Reply:
Done in https://github.com/paraglider-project/paraglider/pull/294/commits/d4c44242aee720ccd3a8478c4c88be751fd3e9c8